apps/web/src/components/credentials/AddCredentialDialog.tsx

```diff
@@ -0,0 +1,325 @@
+"use client";
+
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Textarea } from "@/components/ui/textarea";
+import type { CredentialType } from "@/client/types.gen";
+
+interface AddCredentialDialogProps {
+  onAdd: (credential: {
+    name: string;
+    credential_type: CredentialType;
+    credential_data: Record<string, string>;
+  }) => Promise<void>;
+}
+
+const CREDENTIAL_TYPES: { value: CredentialType; label: string }[] = [
+  { value: "api_key", label: "API Key" },
+  { value: "oauth2", label: "OAuth2" },
+  { value: "basic_auth", label: "Basic Auth" },
+  { value: "token", label: "Token" },
+  { value: "smtp", label: "SMTP" },
+  { value: "custom", label: "Custom" },
+];
+
+// Field configurations for each credential type
+const CREDENTIAL_FIELDS: Record<
+  CredentialType,
+  Array<{
+    key: string;
+    label: string;
+    type: "text" | "password" | "textarea";
+    placeholder?: string;
+    required?: boolean;
+  }>
+> = {
+  api_key: [
+    {
+      key: "api_key",
+      label: "API Key",
+      type: "password",
+      placeholder: "sk_live_...",
+      required: true,
+    },
+    {
+      key: "base_url",
+      label: "Base URL (optional)",
+      type: "text",
+      placeholder: "https://api.example.com",
+    },
+  ],
+  oauth2: [
+    {
+      key: "client_id",
+      label: "Client ID",
+      type: "text",
+      required: true,
+    },
+    {
+      key: "client_secret",
+      label: "Client Secret",
+      type: "password",
+      required: true,
+    },
+    {
+      key: "auth_url",
+      label: "Authorization URL",
+      type: "text",
+      placeholder: "https://oauth.example.com/authorize",
+    },
+    {
+      key: "token_url",
+      label: "Token URL",
+      type: "text",
+      placeholder: "https://oauth.example.com/token",
+    },
+  ],
+  basic_auth: [
+    {
+      key: "username",
+      label: "Username",
+      type: "text",
+      required: true,
+    },
+    {
+      key: "password",
+      label: "Password",
+      type: "password",
+      required: true,
+    },
+  ],
+  token: [
+    {
+      key: "token",
+      label: "Token",
+      type: "password",
+      placeholder: "Bearer token or access token",
+      required: true,
+    },
+  ],
+  smtp: [
+    {
+      key: "host",
+      label: "SMTP Host",
+      type: "text",
+      placeholder: "smtp.gmail.com",
+      required: true,
+    },
+    {
+      key: "port",
+      label: "Port",
+      type: "text",
```

Comment:
```suggestion
      type: "number",
```
Ports are numeric-only values.